### DISCUSSION TITLE
Add option to make MODEL.dict() jsonable/ custom encoding for values

### DISCUSSION THREAD

**Discussion by 07pepa** [pydantic/pydantic#3910] [Feature Request]
sometimes where you do not control whole stack (aws lambda) 
a conversion to json string would be waste and custom encoding may be unvieldy

simple example:
```python
from pydantic import BaseModel
from datetime import datetime,date,timedelta
class MyEvent(BaseModel):
    e:datetime
    f:date
    g:timedelta

def lambda_handler(event,_):
    return MyEvent(**event).dict()
    #this will not work
```
ideal state would be if i could call ` MyEvent(**event).dict(jsonable=True)`
or ` return MyEvent(**event).dict(encode_lut={datetime,lambda d: "stringify it"})` and everything that is extending pydantic models could not be encoded

**Comment by 07pepa** [comment]
will be likely solved by https://github.com/samuelcolvin/pydantic/pull/3346

**Comment by samuelcolvin** [comment]
This is a very old request see #317.

It will hopefully be added some day, but not immediately.

**Reply by 07pepa** [reply]
Thanks for reaching out and next time i will try to search the discussion better... And hopefully some day we will see this feature... Until then I can serialize values in dicts myself

**Reply by samuelcolvin** [reply]
Hopefully it can go into v2. Kind of depends if we move serialization logic into pydantic-core then or later.

Currently just trying to get all validation to work in pydantic-core.
